taxi/tests/test_views.py

```diff
@@ -0,0 +1,129 @@
+
+
+class PublicCarViewsTests(TestCase):
+
+    def test_login_required_car_list(self):
+        response = self.client.get(CAR_LIST_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_car_detail(self):
+        response = self.client.get(CAR_DETAIL_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PublicDriverViewsTests(TestCase):
+
+    def test_login_required_driver_list(self):
+        response = self.client.get(DRIVER_LIST_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_driver_detail(self):
+        response = self.client.get(DRIVER_DETAIL_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivatManufacturerViewsTests(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.user = get_user_model().objects.create_user(
+            username="test", password="<PASSWORD>"
+        )
+        self.client.force_login(self.user)
+        Manufacturer.objects.create(
+            name="Test Manufacturer",
+            country="United Kingdom",
+        )
+        Manufacturer.objects.create(name="Test2 Manufacturer", country="USA")
+        self.response = self.client.get(MANUFACTURER_LIST_URL)
+
+    def test_login_required_manufacturer_list(self):
+        self.assertEqual(self.response.status_code, 200)
+
+    def test_retrieve_manufacturer_list(self):
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(self.response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(self.response, "taxi/manufacturer_list.html")
+
+
+class PrivatCarViewsTests(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.user = get_user_model().objects.create_user(
+            username="test", password="<PASSWORD>"
+        )
+        self.client.force_login(self.user)
+        manufacturer = Manufacturer.objects.create(
+            name="Test Manufacturer", country="USA"
+        )
+        Car.objects.create(
+            model="Toto",
+            manufacturer=manufacturer,
+        )
+        Car.objects.create(
+            model="Polo",
+            manufacturer=manufacturer,
+        )
+        self.response = self.client.get(CAR_LIST_URL)
+
+    def test_login_required_car_list(self):
+        self.assertEqual(self.response.status_code, 200)
+
+    def test_retrieve_car_list(self):
+        response = self.client.get(CAR_LIST_URL)
+        cars = Car.objects.all()
+        self.assertEqual(list(response.context["car_list"]), list(cars))
+        self.assertTemplateUsed(response, "taxi/car_list.html")
+
+    def test_login_required_car_detail(self):
+        response = self.client.get(CAR_DETAIL_URL)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/car_detail.html")
+
+
+class PrivatDriverViewsTests(TestCase):
```

Comment:
The class name `PrivatDriverViewsTests` should be `PrivateDriverViewsTests` to maintain consistency and correct spelling.